SoftLayer/managers/image.py

```diff
@@ -187,3 +188,57 @@ def export_image_to_uri(self, image_id, uri, ibm_api_key=None):
             }, id=image_id)
         else:
             return self.vgbdtg.copyToExternalSource({'uri': uri}, id=image_id)
+
+    def add_locations(self, image_id, location_names):
+        """Add available locations to an archive image template.
+
+        :param int image_id: The ID of the image
+        :param location_names: Locations for the Image.
+        """
+        locations = self.get_locations_id_list(image_id, location_names)
+        locations_ids = [{'id': location_id} for location_id in locations]
+        return self.vgbdtg.addLocations(locations_ids, id=image_id)
+
+    def remove_locations(self, image_id, location_names):
+        """Remove available locations from an archive image template.
+
+        :param int image_id: The ID of the image
+        :param location_names: Locations for the Image.
+        """
+        locations = self.get_locations_id_list(image_id, location_names)
+        locations_ids = [{'id': location_id} for location_id in locations]
+        return self.vgbdtg.removeLocations(locations_ids, id=image_id)
+
+    def get_storage_locations(self, image_id):
+        """Get available locations for public image storage.
+
+        :param int image_id: The ID of the image
+        """
+        return self.vgbdtg.getStorageLocations(id=image_id)
+
+    def get_locations_id_list(self, image_id, location_names):
+        """Converts a list of location names to a list of location IDs.
+
+        :param int image_id: The ID of the image.
+        :param list location_names: A list of location names strings.
+        :returns: A list of locations IDs associated with the given location
+                  keynames in the image id.
+        """
+        locations = self.get_storage_locations(image_id)
+        locations_ids = []
+        matching_location = {}
+        output_error = "Location {} does not exist for available locations for image {}"
+
+        for location_name in location_names:
+            for location in locations:
+                if location_name == location.get('name'):
+                    matching_location = location
+                    break
+            if matching_location.get('id') is None:
+                raise exceptions.SoftLayerError(
+                    output_error.format(location_name, image_id)
+                )
+
+            locations_ids.append(matching_location.get('id'))
```

Comment:
I'd change this to `locations_ids.append(matching_location)` so that you can just pass that list directly to the API and you can remove the `locations_ids = [{'id': location_id} for location_id in locations]` lines in your other two functions.